src/Compiler/SyntaxTree/SyntaxTree.fs

```diff
@@ -775,10 +775,7 @@ type SynExpr =
 
     member e.Range =
         match e with
-        | SynExpr.Paren(_, leftParenRange, rightParenRange, r) ->
-            match rightParenRange with
-            | Some rightParenRange when leftParenRange.FileIndex <> rightParenRange.FileIndex -> leftParenRange
-            | _ -> r
+        | SynExpr.Paren(range = m)
```

Comment:
Is this an intentional change?
Seems unrelated to `#line` from afar.

Reply:
Thanks for reviewing!

This check is no longer necessary. It was there because of #line directives within parenthesized expressions (there are lots of them in the compiler's `pars.fs`). With this PR, however, the AST ranges are always the original ranges and no longer jump around (and we no longer have any `leftParenRange.FileIndex <> rightParenRange.FileIndex` while parsing a single file) , so we can remove this check.

(So, this is just another example of why fixing this issue makes sense.)

Reply:
Sweet, thanks for confirming!